### PR TITLE
Fix #10416  Do not copy ee9 response headers

### LINKED ISSUE #10416: EE9 Copies HttpFields in response
**Jetty version(s)**
12.0.0

**Jetty Environment**
ee8, ee9

**Description**

Discovered in #10359 is that EE9 currently copies the response headers, so that persistent headers like `Server` and `Date` are not well handled.  There as also some TODOs regarding the efficiency of the copy.

### PR DESCRIPTION
**Pull request by gregw**
Use the core response HttpFields directly as the ee9 response headers to avoid copy and retain persistent field behaviour.

### CODE REVIEW

**Review by janbartel — APPROVED** [review]
Seems fine. Is it worthwhile having a  test for setting the "Server" header, which is what revealed this problem in the first place?

**Comment by gregw** [comment]
@janbartel can you re-review... @lachlan-roberts can you review.
I've changed the fix for #10339 now so that when response fields are cleared they reset to their original values rather than keep their modified values.

File: jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java [hunk 1]
```diff
@@ -1272,25 +1277,7 @@ public void setTrailerFields(Supplier<Map<String, String>> trailers)
...
-     * @return The committed MetaData or a {@link #newResponseMetaData()}
-     * if not yet committed.
-     */
-    public MetaData.Response getCommittedMetaData()
-    {
-        MetaData.Response meta = _channel.getCommittedMetaData();
-        if (meta == null)
-            return newResponseMetaData();
-        return meta;
+        return new MetaData.Response(getStatus(), getReason(), _channel.getRequest().getHttpVersion(), _channel.getCoreResponse().getHeaders(), getLongContentLength(), getTrailers());
```
Comment (lachlan-roberts) [inline jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java:1280]: Why are you using `_channel.getCoreResponse().getHeaders()` and not `_fields`?

**Review by janbartel — APPROVED** [review]

**Review by lorban — APPROVED** [review]
LGTM

File: jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java (hunk 1 above)
Comment (gregw) [reply jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java:1280]: They should now be the same instance.... but if somehow they are not the same, then we should use the ones form the core request.  I guess I could put in:
```suggestion
        assert _channel.getCoreResponse().getHeaders() == _fields
        return new MetaData.Response(getStatus(), getReason(), _channel.getRequest().getHttpVersion(), _channel.getCoreResponse().getHeaders(), getLongContentLength(), getTrailers());
```
But currently there is no way for that to not be true, so I'm good leaving it out unless you think it makes it clearer.

Comment (lachlan-roberts) [reply jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java:1280]: Yeah it just looks weird that all the modifications are done on the `_fields` `HttpFields.Mutable` which is passed in. But then when it is used we just rely on an external assumption that it is the one from the `getCoreResponse()`.

Do you even need `protected void onResponse(HttpFields.Mutable headers)` method when you could just assign `_fields = _channel.getCoreResponse().getHeaders()`?

**Review by joakime — APPROVED** [review]

**Pull request merged**